### DISCUSSION TITLE
I ported ORT into Pascal ( FreePascal and Delphi ) and it works like a charm!

### DISCUSSION THREAD

**Discussion by hshatti** [microsoft/onnxruntime#15543] [Ideas / Feature Requests]
Hi, i just loved working with ONNX Runtime and had a thought of porting it into Object Pascal/Lazarus or Delphi and making it available for these old but gold languages for fun, it wasn't a short journey though but it finally worked. 

Why? I had these goals in mind:
- Fun! 😃 
- Modern pascal environment has the reputation of being powerful enough to develop desktop applications with a built-in support for almost any sort of known databases, along side with built-in image manipulation libraries and other native libraries, so bringing ORT into it would be a perfect accessory.
- Making use of the rich and powerful GUI that freepascal offers ( or the modern FMX GUI provided by Delphi ) and incorporate ORT with it out of the box.
- Single executable that can be deployed with onnxruntime libraries (minimal ORT release for CPU comes with 2 files or may be it's not even required since they come shipped with modern windows releases AFAIK) no other dependencies are needed, 

I made an example that can run on single executable (windows or MacOS) 
if anyone interested or any of the ORT team find it a useful contribution please feel free to check or review and let me know, or to incorporate the new languages implementation to the family of languages that ORT can now support.

## The repository :
https://github.com/hshatti/TONNXRuntime

Contributions, review and criticism are most welcome.
Cheers 
H

**Comment by VasProgger** [comment]
cool! thanks!
